course/src/main/java/org/openedx/course/presentation/calendarsync/CalendarManager.kt

```diff
@@ -0,0 +1,396 @@
+     */
+    fun deleteCalendar(calendarId: Long) {
+        context.contentResolver.delete(
+            Uri.parse("content://com.android.calendar/calendars/$calendarId"),
+            null,
+            null
+        )
+    }
+
+    /**
+     * Helper method used to return a URI for use with a sync adapter (how an application and a
+     * sync adapter access the Calendar Provider)
+     *
+     * @param uri URI to access the calendar
+     * @param account Name of the calendar owner
+     *
+     * @return URI of the calendar
+     *
+     */
+    private fun asSyncAdapter(uri: Uri, account: String): Uri? {
+        return uri.buildUpon().appendQueryParameter(CalendarContract.CALLER_IS_SYNCADAPTER, "true")
+            .appendQueryParameter(CalendarContract.SyncState.ACCOUNT_NAME, account)
+            .appendQueryParameter(
+                CalendarContract.SyncState.ACCOUNT_TYPE,
+                CalendarContract.ACCOUNT_TYPE_LOCAL
+            ).build()
+    }
+
+    fun openCalendarApp() {
+        val builder: Uri.Builder = CalendarContract.CONTENT_URI.buildUpon()
+            .appendPath("time")
+        ContentUris.appendId(builder, Calendar.getInstance().timeInMillis)
+        val intent = Intent(Intent.ACTION_VIEW).setData(builder.build())
+        intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
+        context.startActivity(intent)
+    }
+
+    /**
+     * Helper method used to check that the calendar if outdated for the course or not
+     *
+     * @param calendarTitle Title for the course Calendar
+     * @param courseDateBlocks Course dates events
+     *
+     * @return Calendar Id if Calendar is outdated otherwise -1 or CALENDAR_DOES_NOT_EXIST
+     *
+     */
+    fun isCalendarOutOfDate(
+        calendarTitle: String,
+        courseDateBlocks: List<CourseDateBlock>
+    ): Long {
+        if (isCalendarExists(calendarTitle)) {
+            val calendarId = getCalendarId(calendarTitle)
+            if (compareEvents(calendarId, courseDateBlocks).not()) {
+                return calendarId
+            }
+        }
+        return CALENDAR_DOES_NOT_EXIST
+    }
+
+    /**
+     * Method to get the current user account as the Calendar owner
+     *
+     * @return calendar owner account or "local_user"
+     */
+    private fun getUserAccountForSync(): String {
+        return corePreferences.user?.email ?: LOCAL_USER
+    }
+
+    /**
+     * Method to create the Calendar title for the platform against the course
+     *
+     * @param courseName Name of the course for that creating the Calendar events.
+     *
+     * @return title of the Calendar against the course
+     */
+    fun getCourseCalendarTitle(courseName: String): String {
+        return "${resourceManager.getString(id = CoreR.string.platform_name)} - $courseName"
+    }
+
+    companion object {
+        const val CALENDAR_DOES_NOT_EXIST = -1L
+        private const val REMINDER_24_HOURS = 24 * 60
```

Comment:
we can use TimeUnit for this, for example: TimeUnit.DAYS.toMinutes(1) or TimeUnit.HOURS.toMinutes(24)